sdk/eventhub/event-hubs/test/receiver.spec.ts

```diff
@@ -292,6 +292,126 @@ describe("EventHub Receiver", function(): void {
     });
   });
 
+  describe("getIterator", function(): void {
+    it("should receive messages correctly", async function(): Promise<void> {
+      const partitionId = partitionIds[0];
+
+      const time = Date.now();
+      // ensure messages exist to retrieve
+      const messageCount = 5;
+      const sentEventData: EventData[] = [];
+      for (let i = 0; i < messageCount; i++) {
+        sentEventData.push({
+          body: `getIterator test message ${Date.now()} - ${i}`
+        });
+      }
+
+      const sender = client.createSender({partitionId: partitionId});
+      try {
+        await sender.send(sentEventData);
+      } finally {
+        await sender.close();
+      }
+
+      receiver = client.createReceiver(partitionId, {
+        eventPosition: EventPosition.fromEnqueuedTime(time)
+      });
+      const eventIterator = receiver.getEventIterator();
+
+      let messagesReceivedCount = 0;
+      const data: ReceivedEventData[] = [];
+      for await (const event of eventIterator) {
+        data.push(event);
+        if (++messagesReceivedCount >= messageCount) {
```

Comment:
Say we got the first event and are waiting for the second.
For any which reason, if the underlying receiver does not get the second event, then we will never exit this for loop and the test would time out, correct?

Instead, why not pass in a cancellation token which cancels the operation after a particular time period?

Reply:
There's an issue with passing the cancellation token with the event iterator (issue #3597) that causes an unhandled exception when you cancel, so I think that needs to be fixed before refactoring this. But you're right in that I'm currently relying on the test to timeout which can vary depending on what the test runner passes as the value.